File: src/main/java/net/hectus/neobb/turn/default_game/item/ItemTurn.java
package net.hectus.neobb.turn.default_game.item;

import net.hectus.neobb.player.NeoPlayer;
import net.hectus.neobb.turn.Turn;
import net.hectus.neobb.util.Utilities;
import org.bukkit.Location;
import org.bukkit.Material;
import org.bukkit.inventory.ItemStack;

public abstract class ItemTurn extends Turn<ItemStack> {
    public ItemTurn(NeoPlayer player) { super(null, null, player); }
    public ItemTurn(ItemStack data, Location location, NeoPlayer player) { super(data, location, player); }

    @Override
    public ItemStack item() {
        return new ItemStack(Material.valueOf(Utilities.camelToSnake(Utilities.counterFilterName(getClass().getSimpleName())).toUpperCase()));
    }
}

File: src/main/kotlin/net/hectus/neobb/modes/turn/default_game/item/ItemTurn.kt
package net.hectus.neobb.modes.turn.default_game.item

import com.marcpg.libpg.storing.Cord
import net.hectus.neobb.modes.turn.Turn
import net.hectus.neobb.player.NeoPlayer
import net.hectus.neobb.util.material
import org.bukkit.inventory.ItemStack

abstract class ItemTurn(data: ItemStack?, cord: Cord?, player: NeoPlayer?) : Turn<ItemStack>(data, cord, player) {
    override fun item(): ItemStack = ItemStack(this::class.material())
}
